datasketches-memory-java8/src/main/java/org/apache/datasketches/memory/Memory.java

```diff
@@ -196,10 +208,12 @@ static Memory wrap(byte[] array, ByteOrder byteOrder) {
    * @param byteOrder the byte order to be used.  It must be non-null.
    * @return a new <i>Memory</i> for read operations
    */
-  static Memory wrap(byte[] array, int offsetBytes, int lengthBytes, ByteOrder byteOrder) {
-    Objects.requireNonNull(array, "array must be non-null");
-    ResourceImpl.checkBounds(offsetBytes, lengthBytes, array.length);
-    return BaseWritableMemoryImpl.wrapHeapArray(array, offsetBytes, lengthBytes, true, ByteOrder.nativeOrder(), null);
+  static Memory wrap(
+      byte[] array, 
+      int offsetBytes, 
+      int lengthBytes, 
+      ByteOrder byteOrder) {
+    return BaseWritableMemoryImpl.wrapHeapArray(array, offsetBytes, lengthBytes, true, byteOrder, null);
```

Comment:
This fixes a subtle bug that has been there since 2.0.0: The user-specified ByteOrder was overridden with ByteOrder.nativeOrder().  -- This would have affected the BO on subsequent multi-byte method calls only if the user specified a non-native BO.